plugins/outputs/bigquery/bigquery.go

```diff
@@ -0,0 +1,233 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"sync"
+	"time"
+
+	"cloud.google.com/go/bigquery"
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/outputs"
+)
+
+const (
+	defaultOffSetKey   = "offset-key.json"
+	timeStampFieldName = "timestamp"
+)
+
+var defaultTimeout = 5 * time.Second
+
+const sampleConfig = `	
+  ## Credentials File
+  credentials_file = "/path/to/service/account/key.json"
+  ## GCP Project
+  project = "my-gcp-project"
+
+  ## The namespace for the metric descriptor
+  dataset = "telegraf"
+
+  ## Timeout for BigQuery operations.  Minimum timeout is "1s".
```

Comment:
Is there a reason you enforce a timeout of one second? If it is about default-value handling, the easy solution is to set the default in `init()` when initializing the structure. This way the value is only overwritten when set by the user.

Reply:
Hi @srebhan no specific reason. Was just checking the zookeeper connector that did the same. Did the default as you proposed.